packages/app-project/src/screens/ClassifyPage/components/YourStats/YourStats.spec.js

```diff
@@ -1,15 +1,41 @@
-import { shallow } from 'enzyme'
+import { render, screen } from '@testing-library/react'
+import { composeStory } from '@storybook/react'
+import Meta, { YourStats } from './YourStats.stories.js'
+import { YourStatsStoreMock } from './YourStats.mock'
 
-import YourStats from './YourStats'
+describe('Component > YourStats', function () {
+  beforeEach(function () {
+		const YourStatsStory = composeStory(YourStats, Meta)
+		render(<YourStatsStory />)
+	})
 
-let wrapper
+	it('should have a heading', function () {
+    expect(screen.findByText(`Your ${YourStatsStoreMock.project.display_name} Statistics`)).to.be.ok()
+  })
 
-describe('Component > YourStats', function () {
-  before(function () {
-    wrapper = shallow(<YourStats />)
+	it('should have today\'s classification count ', function () {
+    expect(screen.getAllByText(`${YourStatsStoreMock.user.personalization.counts.today}`).length).to.equal(2)
+  })
+
+	it('should have this week\'s classification total ', function () {
+    expect(screen.queryByText(`${YourStatsStoreMock.user.personalization.counts.total}`)).to.be.ok()
+  })
```

Comment:
**Note 2:** this test actually fails for me with the error message, "TestingLibraryElementError: Found multiple elements with the text: 100"

This seems pretty straightforward - the text that we're looking for is _too generic._ The 100 can describe either the classifications total (which is what we're looking for), or a number on the y-axis. 🤷 

<img width="534" alt="image" src="https://github.com/zooniverse/front-end-monorepo/assets/13952701/a7f36ff8-dfca-4a95-8c03-d1ca7d95ebdd">

I think a better implementation is to be _very specific_ about what we're supposed to be looking at. i.e. "Find the text next to the text 'Classifications Total', and check that the text says '100'."

However, as to _how_ you'd create such a specific query... erm... gimme a second. Processing, processing...

Reply:
Additionally: simply replacing `expect(getByText('100')).to.be.ok()` with `expect(getAllByText('100')).to.equal(2)` isn't a great solution either, as it's just looking around the rendered screen for the existence of the number 100, without care of the context of how/where that number is used.

Anyway, back to processing a suggested solution...

Reply:
`within` is your friend in situations like these, or chaining RTL queries from a previous query result.

Reply:
Eg. 

https://github.com/zooniverse/front-end-monorepo/blob/ee66328323a5fb37245f027fad8154bedbe77876/packages/lib-classifier/src/components/Classifier/Classifier.spec.js#L88

Reply:
**Suggested Solution**

OK, so we need to be very specific in our query to find that "100 Classifications Total" element.

Option 1: find the text "Classifications Total", then find the "100"

From studying the HTML, we know that the "Classifications Total" text sits in its own div, and the number "100" exists in the div juuust before it. So...

```
it('should have this week\'s classification total ', function () {
  const elem = screen.getByText('Classify.YourStats.totalCount')
  expect(elem.previousElementSibling.textContent).to.equal('100')
})
```

Option 2: add classnames to the stats, so we can easily distinguish the element via code

In YourStats.js, add some descriptive class names:
```
<Stat
  className='your-stats_total'
  label={t('Classify.YourStats.totalCount')}
  value={counts.total}
/>
```

In YourStats.spec.js, look for the element with the classname, then use ['within'](https://testing-library.com/docs/dom-testing-library/api-within/) to query _inside that element_
```
import { within } from '@testing-library/dom'

...

it('should have this week\'s classification total ', function () {
  const elem = document.getElementsByClassName('your-stats_total')[0]
  expect(within(elem).getByText('100')).to.be.ok()
})
```

Reply:
This was a bit more complicated because the `Stat` component does not allow setting data attrbiutes... I ended up going with the class route wherein the classes are named `test-classify-your-stats-today-count` and `test-classify-your-stats-total-count` to indicate that they are used for a testing instead of stylistic purpose.

I'm not a big fan of going too deep into complex querying the internal structure of a component (and in this case a sub-component) for the same reasons RTL doesn't have a getByClass() method... Which is its focused on implementation structure. That all said, I think this is a reasonable path with which to move forward!